shelley/chain-and-ledger/executable-spec/src/BlockChain.hs

```diff
@@ -122,10 +130,9 @@ instance ToCBOR ProtVer where
 data BHBody hashAlgo dsignAlgo kesAlgo = BHBody
   { -- | Hash of the previous block header
     -- The first block in a chain will set this field to Nothing.
-    -- TODO Since the Shelley chain will begins with blocks from
-    -- the Byron era, we should probably use a sum type here,
-    -- so that the first shelley block can point to the last Byron block.
-    bheaderPrev           :: Maybe (HashHeader hashAlgo dsignAlgo kesAlgo)
+    bheaderPrev           :: Either
```

Comment:
it would of course be nice to reflect that the `Left` type can only appear once. We should probably make at least a property for this

Reply:
Also, shouldn't we update this in the Shelley spec?

Reply:
yea, that's a good property.

I am of the opinion that we should just keep it as a generic hash in the formal spec. Even in the exec model both headers hashes are of type `Hash`, it's just that there is a phantom type. But I'm happy to change the formal spec to reflect this if you and/or other think otherwise!